### ISSUE TITLE
Model not compiling

### ISSUE THREAD

**Issue by aysnrarsln** [mjposs/min-max-min#4]
https://github.com/mjposs/min-max-min/blob/625759e484c0ad776036b8e97e371acd8af7855e/Applications/KP.jl#L306

This line does not compile. Throws the following error message

LoadError: LoadError: syntax: expected "}" or separator in arguments to "{ }"; got "(η ≤ sum(((((1 + (sum(((data.Φ[i, j] * ξ[j]) for j = 1:data.M)) / 2)) * data.hc[i]) * x[k, i]) for i = N))))"

Are you sure that we can use logical implication syntax like this? I am using Julia 1.2.0 by the way, maybe this is syntax from 1.4.0? Can you confirm that it compiles on your computer?

**Comment by mjposs** [comment]
En effet, il y avait une parenthèse en trop. Voir les détails ici: 
https://jump.dev/JuMP.jl/stable/constraints/#Indicator-constraints

**Comment by mjposs** [comment]
(J'ai pushé la correction)

**Comment by aysnrarsln** [comment]
Ok, ça passe! Par contre, L n'est pas défini dans la partie juste après. C'est data.numconst?

**Comment by mjposs** [comment]
Tout à fait. Peut être changer ncons par L pour être plus proche du pdf!

**Comment by mjposs** [comment]
Mais si tu fais le changement, bien le reporter dans tout algorithm.jl car c'est utilisé souvent.

**Comment by aysnrarsln** [comment]
Pour le moment je mets L = data.ncons localement dans la separation. Par contre, les variables sont indexés à partir de 0 et donc Julia n'aime pas. Mon instinct initial est d'augmenter les indices par 1 mais faut pe que je fasse plus gaffe... Je vais regarder ça

**Comment by mjposs** [comment]
> Pour le moment je mets L = data.ncons localement dans la separation. Par contre, les variables sont indexés à partir de 0 et donc Julia n'aime pas. Mon instinct initial est d'augmenter les indices par 1 mais faut pe que je fasse plus gaffe... Je vais regarder ça
> 
Pq Julia n'aime pas ? Si ce n'est pas un bête array, ça devrait aller non ? Enfin je pensais ...
Mais sinon, il vaut déjà mieux mettre le 0 en L+1, pour ne pas affecter les nombres intermédiaires. Ca évitera des erreurs.

**Comment by aysnrarsln** [comment]
Finalement c'est pe parce que ta variable z était definie comme z[l] mais utilisée comme z[k,l]. En tout cas, je change les indices pour etre dans la convention de Julia...

**Comment by mjposs** [comment]
Envoyé de mon iPad

> Le 26 mars 2021 à 21:33, aysnrarsln ***@***.***> a écrit :
> 
> ﻿
> Finalement c'est pe parce que ta variable z était definie comme z[l] mais utilisée comme z[k,l]. En tout cas, je change les indices pour etre dans la convention de Julia...
> 
Je le ferais pas: pour les modèles MILP je préfère toujours être aussi proche que possible du PDF, ça facilite la lecture et évite les erreurs. Mais bon, c'est juste mon avis. En tous cas si tu mets 0 <-> L+1, il faut le mettre dans les commentaires à mon avis.

**Comment by aysnrarsln** [comment]
ca marche, la je l'ai changé mais je reviendrai en arrière demain. Par contre, l'algo ne marche plus. J'ai le meme probleme que ce matin: un boucle infinie on dirait, ça affiche rien après heuristique mode du warmup. J’enquête ça demain également...
